✨ Source Github: add `files` and `stats` propertied to `pull_request_commits` schema

dlouseiro opened the pull request:
## What

The current implementation of the `commits` stream in the source Github is missing a property called `files`, documented in the response schema of Github's REST API for commits ([here](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits)).

This property is an array, containing detailed information on the changes done in each file within the commit (number of additions, number of removals, if the file was added or removed, etc).

This PR resolves #34318 

## How

Added a simplified version of the response schema provided by Github in `airbyte-integrations/connectors/source-github/source_github/schemas/commits.json`.

## 🚨 User Impact 🚨

As this PR simply adds a property to an existing stream, it should correspond to a minor version of the connector.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/airbytehq/airbyte?pullRequest=34317) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/airbytehq/airbyte?pullRequest=34317) before we can accept your contribution.<br/><hr/>**Dinis Louseiro** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/airbytehq/airbyte?pullRequest=34317) it.</sub>

dlouseiro commented:
Resolved #34317

marcosmarxm commented on airbyte-integrations/connectors/source-github/source_github/schemas/commits.json:
```diff
@@ -121,6 +121,47 @@
+      "type": ["null", "array"],
+      "items": {
+        "type": ["null","object"],
+        "properties": {
```
@dlouseiro it's better to add the `additionalProperties: true` so in cases there is a missing or new property the connector won't break.

dlouseiro replied:
Sure, makes sense! Will do

marcosmarxm reviewed (changes requested):
Small detail, can you bump the metadata.yaml and changelog version in the doc? Take a look in the guidelines here: https://docs.airbyte.com/contributing-to-airbyte/change-cdk-connector#code-your-contribution

dlouseiro commented:
> Small detail, can you bump the metadata.yaml and changelog version in the doc? Take a look in the guidelines here: https://docs.airbyte.com/contributing-to-airbyte/change-cdk-connector#code-your-contribution

Sure, will do!

dlouseiro commented:
@marcosmarxm , in the meantime I did some extra tests and noticed that in order to fetch the `files` property for the `commits` stream we'd need to do it [commit by commit](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#get-a-commit) as this property is not available in the API to [list commits](https://docs.github.com/en/rest/commits/commits?apiVersion=2022-11-28#list-commits).

Although, the same `files` as well as `stats` properties are available when fetching [Pull request commits](https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request) so I removed them from the `commits` stream and added them to the `pull_request_commits` stream.

lazebnyi commented:
@dlouseiro Thanks for your PR. But pull requests commits schema has not field files - https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request

dlouseiro commented:
> @dlouseiro Thanks for your PR. But pull requests commits schema has not field files - https://docs.github.com/en/rest/pulls/pulls?apiVersion=2022-11-28#list-commits-on-a-pull-request 

hey @lazebnyi youre completely right. I meant to close this pr in the meantime. The files and stats properties are listed in the response schema of the list pr commits API, but are actually not sent in the payload.

I based the or on the response schema documented by GitHub but when tried to test it did not receive the files not stats properties. Thanks for closing it
